server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,590 @@
+            else:
+                print(
+                    f"{item_path} item should have been handled by the above mentioned conditions. "
+                    f"It is an unexpected item which should not have occured, "
+                    f"leading to an inappropriate condition like a hidden directory."
+                )
+        verify_subdirs(hier, controller, controller_subdir)
+        verify_prefixes(hier, controller)
+
+    return 0
+
+
+# Incoming Hierarchy
+def verify_tar_dirs(ihier, tarball_dirs, tblist, controller):
+    for tb in tarball_dirs:
+        if tb.endswith("unpack"):
+            tar = tb[:-7]
+            tar = f"{tar}.tar.xz"
+            val = Hierarchy.INVALID_UNPACKING_DIRS
+        else:
+            tar = f"{tb}.tar.xz"
+            val = Hierarchy.INVALID_TB_DIRS
+        tarfile = Path(ihier.config.ARCHIVE, controller, tar)
+        if tarfile.exists():
+            with tarfile.open(mode="r") as f:
+                try:
+                    file_content = f.read(1)
+                    if file_content:
+                        continue
+                except Exception:
+                    pass
+        else:
+            tblist(val, controller, tb)
+    return 0
+
+
+# 'ihier' is IncomingHierarchy class object
+def verify_incoming(ihier, verifylist):
+
+    for controller in verifylist:
+        ihier.add_controller(controller)
+        direct_entries = Path(ihier.config.INCOMING, controller).glob("*")
+        tarball_dirs = list()
+        unpacking_tarball_dirs = list()
+
+        if not Path(ihier.config.ARCHIVE, controller).is_dir():
+            # Skip incoming controller directories that don't have an $ARCHIVE
+            # directory, handled in another part of the audit.
+            continue
+
+        for dir_path in direct_entries:
+            if (
+                dir_path.is_dir()
+                and not dir_path.name.endswith(".unpack")
+                and not len(os.listdir(dir_path)) == 0
+            ):
+                tarball_dirs.append(dir_path.name)
+            elif (
+                dir_path.is_dir()
+                and not dir_path.name.endswith(".unpack")
+                and len(os.listdir(dir_path)) == 0
+            ):
+                ihier.add_unexpected_controllers(
+                    Hierarchy.EMPTY_TARBALL_DIRS, controller, dir_path.name
+                )
+            elif (
+                dir_path.is_dir()
+                and dir_path.name.endswith(".unpack")
+                and len(os.listdir(dir_path)) == 0
+            ):
+                unpacking_tarball_dirs.append(dir_path.name)
+            elif dir_path.is_symlink():
+                ihier.add_unexpected_controllers(
+                    Hierarchy.TARBALL_LINKS, controller, dir_path.name
+                )
+
+        if tarball_dirs:
+            verify_tar_dirs(
+                ihier, tarball_dirs, ihier.add_unexpected_controllers, controller
+            )
+
+        if unpacking_tarball_dirs:
```

Comment:
This should probably be deleted: audit just happened to collide with an unpack - there is nothing that needs to be done about that. If there are problems, we can wait until the next audit invocation to find them: by that time, the unpack will be done - we hope!

Reply:
it is leading to test cases failure. We can discuss this.

Reply:
due to some missing reported unpacking_tarball_dirs